src/components/molecules/ResourceGraphTab/ResourceGraphTab.tsx

```diff
@@ -15,7 +20,13 @@ const ResourceGraphTab: React.FC = () => {
   const activeStorage = useAppSelector(activeResourceStorageSelector);
   const selectedResource = useSelectedResource();
   const resources = useMemo(() => (selectedResource ? [selectedResource] : []), [selectedResource]);
-  const resourceMap = useResourceMap(activeStorage);
+  const activeResoureMetaMap = useActiveResourceMetaMap();
+  const activeResoureContentMap = useActiveResourceContentMap();
+  // TODO: computing this is expensive, but the Graph is from core and it needs the resource map...
+  const resourceMap = useMemo(
+    () => joinK8sResourceMap(activeResoureMetaMap, activeResoureContentMap),
```

Comment:
Could we somehow compute this while creating the active resource meta/content map?

Reply:
Getting the active meta/content map is just returning the correct map based on the current active storage, not doing anything else. I'm creating the resourceMap here only because the Graph needs it, in all other places we should try to just not create that map and use the separate meta and content maps..